cache/fast_lru_cache.h

```diff
@@ -26,26 +35,39 @@ namespace fast_lru_cache {
 struct LRUHandle {
   void* value;
   Cache::DeleterFn deleter;
-  LRUHandle* next_hash;
   LRUHandle* next;
   LRUHandle* prev;
   size_t total_charge;  // TODO(opt): Only allow uint32_t?
-  size_t key_length;
   // The hash of key(). Used for fast sharding and comparisons.
   uint32_t hash;
   // The number of external refs to this entry. The cache itself is not counted.
   uint32_t refs;
 
   enum Flags : uint8_t {
-    // Whether this entry is referenced by the hash table.
-    IN_CACHE = (1 << 0),
+    // Whether the handle can be found via a Lookup.
+    IS_VISIBLE = (1 << 0),
+    // Whether the slot is in use by an element.
+    IS_ELEMENT = (1 << 1),
   };
   uint8_t flags;
 
-  // Beginning of the key (MUST BE THE LAST FIELD IN THIS STRUCT!)
-  char key_data[1];
+  // Number of elements that hash to this slot or a lower one,
+  // but wind up in a higher slot.
+  uint32_t displacements;
+
+  // Every slot in the hash table is an LRUHandle e, which be in three different
+  // states:
+  // - Element (IS_ELEMENT set): The slot contains an actual key-value element.
+  // - Tombstone (IS_ELEMENT unset and displacements > 0): The slot contains a
+  // tombstone.
+  // - Empty (IS_ELEMENT unset and displacements == 0): The slot is unused.
+  // A slot that is an element can further have IS_VISIBLE set or not.
```

Comment:
Which variable is this comment section for?

Reply:
I'll make this a global comment, explaining how the cache works.